### ISSUE TITLE
Params in Flag module return Option type

### ISSUE THREAD

**Issue by EchoPouet** [vlang/v#19235]
### Describe the feature

Hello,

I use `flag` module and I would like that an parameter returns an Option type. The value returned can be the param value or `none`/ `default` value only if defined.

With this feature, We can know if a parameter is used because with default value it's not possible.

### Use Case

Exemple:
```
a_bool := fp.bool('a_bool', 0, none, 'some boolean flag. --a_bool will set it to true.')
```

Because default value is none, `a_bool` value will be `none` or true/false.

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### Version used

0.0.4

### Environment details (OS name and version, etc.)

Windows 11

**Comment by yuyi98** [comment]
Does `bool_opt(...)` meet your requirements?

**Comment by EchoPouet** [comment]
You are right but why return an error ? By principal, If it's optional it's not an error if missing.

**Comment by spytheman** [comment]
Errors are more informative than Options. 

Currently we do return error() just to signal that something was not present, but in the future, we could have some flag value that is present, but not valid, and we can then return a more specialized error.

With an Option as the return type, we can not tell that to the user.

I agree, that it can be confusing, why a method that ends with _opt() returns an error, but in this case the _opt means that the presence of that flag on the command line is optional.
